cura/API/Account.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+from typing import Optional, Dict
+
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot, pyqtProperty
+
+from UM.Message import Message
+from cura.OAuth2.AuthorizationService import AuthorizationService
+from cura.OAuth2.Models import OAuth2Settings
+from UM.Application import Application
+
+from UM.i18n import i18nCatalog
+i18n_catalog = i18nCatalog("cura")
+
+
+##  The account API provides a version-proof bridge to use Ultimaker Accounts
+#
+#   Usage:
+#       ``from cura.API import CuraAPI
+#       api = CuraAPI()
+#       api.account.login()
+#       api.account.logout()
+#       api.account.userProfile # Who is logged in``
+#
+class Account(QObject):
+    # Signal emitted when user logged in or out.
+    loginStateChanged = pyqtSignal()
+
+    def __init__(self, parent = None) -> None:
+        super().__init__(parent)
+        self._callback_port = 32118
+        self._oauth_root = "https://account.ultimaker.com"
+        self._cloud_api_root = "https://api.ultimaker.com"
+
+        self._oauth_settings = OAuth2Settings(
+            OAUTH_SERVER_URL= self._oauth_root,
+            CALLBACK_PORT=self._callback_port,
+            CALLBACK_URL="http://localhost:{}/callback".format(self._callback_port),
+            CLIENT_ID="um---------------ultimaker_cura",
+            CLIENT_SCOPES="user.read drive.backups.read drive.backups.write client.package.download",
+            AUTH_DATA_PREFERENCE_KEY="general/ultimaker_auth_data",
```

Comment:
Note: now that this key is changed to something more generic, people that will upgrade to Cura 3.6 (?) and the then updated backups plugin with it, will have to sign in again. Not per se a problem, just noting the implication here.

Reply:
Sorry, I probably put this in the wrong place. I meant something like this for Preferences.